pkg/cluster/cluster.go

```diff
@@ -276,5 +277,25 @@ func (c *Cluster) createServiceAccount() error {
 
 // teardownNamespace tears down the cluster namespace
 func (c *Cluster) teardownNamespace() error {
-	return c.client.CoreV1().Namespaces().Delete(c.namespace, &metav1.DeleteOptions{})
+	step := logging.NewStep(c.namespace, "Delete namespace %s", c.namespace)
+	step.Start()
+	err := c.client.CoreV1().Namespaces().Delete(c.namespace, &metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	w, err := c.client.CoreV1().Namespaces().Watch(metav1.ListOptions{
```

Comment:
It looks to me like there's a potential race condition here. The watch should be added before `Delete`, otherwise the delete could happen before the watch is created and an event may never come.